swan-lake/featured-scenarios/manage-data-persistence-with-bal-persist.md

```diff
@@ -0,0 +1,501 @@
+
+```
+├── rainier
+│   ├── generated
+│   │		└── db
+│   │		│	├── persist_client.bal
+│   │		│	└── persist_types.bal
+│   ├── persist
+│   │		└── model.bal
+│   ├── Ballerina.toml
+│   └── main.bal
+    
+```
+
+The `persist generate` will parse the `persist/model.bal` definition file and generate the following.
+
+|         File          |                                         Description                                         |
+|:---------------------:|:-------------------------------------------------------------------------------------------:|
+| `persist_client.bal`  |               This is the client that is used to persist and retrieve data..                |
+|  `persist_types.bal`  |       This contains the record types that will be used to persist and retrieve data.        |
+
+
+
+> Note: All of the above auto-generated bal files should not be modified.
+
+> Note: If you use other data stores like MySQL or Google Sheets, additional files are created to handle runtime configurations and scripts to set up the database/worksheet. Since we are using `in-memory`, we don’t need any configuration and pre-setting up to run the application.
+
+## Query your database with Persist Client
+
+You first need to instantiate the generated client object inside the `rainier.store` module like below,
+
+```ballerina
+import rainier.store;
+
+store:Client sClient = check new();
+```
+
+Now you can use this `sClient` as the client to query your database. The client provides five resource methods to query. Let’s explore each of these clients’ methods separately. You can use the `main.bal` file created inside your project to explore the functionalities.
+
+### Create a new `Employee` record
+
+Let’s start with a small query to create a new `Employee` record in the database and log the results to the console. Update the following in your `main.bal` file.
+
+```ballerina
+import ballerina/io;
+import ballerina/uuid;
+import rainier.store;
+
+
+final store:Client sClient = check new();
+
+
+public function main() returns error? {
+   store:EmployeeInsert employee1 = {
+       employee_id: uuid:createType4AsString(),
+       first_name: "John",
+       last_name: "Doe",
+       email: "johnd@xyz.com",
+       phone: "1234567890",
+       hire_date: {
+           year: 2020,
+           month: 10,
+           day: 10
+       },
+       manager_id: "123e4567-e89b-12d3-a456-426614174000",
+       job_title: "Software Engineer"
+   };
+
+
+   string[] employeeIds = check sClient->/employees.post([employee1]);
+   io:println("Inserted employee id: " + employeeIds[0]);
+}
+```
+
+You can run your Ballerina project with the following command:
+
+```
+$ bal run
+Compiling source
+        foo/rainier:0.1.0
+
+
+Running executable
+
+Inserted employee id: 16c6553a-373c-4b29-b1c8-c282f444248c
+```
+
+Congratulations!! You created your first database record with the Persist Client. In the next sections, you will learn how to read data from the database.
```

Comment:
```suggestion
This creates the first database record with the Persist Client. The next sections describe how to read data from the database.
```
Need to avoid marketing jargon in technical docs.